### PR TITLE
fix: remove meta widget from debugger and tern server

### LINKED ISSUE #18382: Task: Remove Meta widgets from autocomplete and debugger
Remove all meta widgets from Autocomplete

Screenshot 2022-11-22 at 10 45 46

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Filtered The data tree to remove meta Widgets(Non-Template) from the tern server and updated the debugger Selector to exclude meta Widgets(Non-Template)

Fixes #18382

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by ohansFavour — COMMENTED** [review]
AFAIK, this is what the privateWidgets functionality also does. Can we completely remove the privateWidgets functionality?

File: app/client/src/sagas/PostEvaluationSagas.ts [hunk 1]
```diff
@@ -372,13 +372,12 @@ export function* updateTernDefinitions(
   }
   if (shouldUpdate) {
     const start = performance.now();
-    // remove private widgets from dataTree used for autocompletion
-    const treeWithoutPrivateWidgets = getDataTreeWithoutPrivateWidgets(
-      dataTree,
-    );
+    // remove private and Non-Template meta widgets from dataTree used for autocompletion
+    const filteredDataTree = getFilteredDataTree(dataTree);
```
[the remark this answers is not in the document]
Comment (Tooluloope) [reply app/client/src/sagas/PostEvaluationSagas.ts:376]: Okay, this is fine.

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 2]
```diff
@@ -465,6 +465,10 @@ class MetaWidgetGenerator {
     metaWidget.children = children;
     metaWidget.parentId = parentId;
     metaWidget.referencedWidgetId = templateWidgetId;
+    if (metaWidget.widgetId !== templateWidgetId) {
+      metaWidget.suppressAutoComplete = true;
+      metaWidget.suppressDebuggerError = true;
+    }
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:471]: You can move this into [this](https://github.com/appsmithorg/appsmith/pull/18656/files#diff-b673b712c3c6608c19af587167a07d79733ee58b94977ee9c6b516328ee5d2e1R458) condition

File: app/client/src/workers/Evaluation/evaluationUtils.ts [hunk 3]
```diff
@@ -706,6 +712,27 @@ export const getDataTreeWithoutPrivateWidgets = (
   const treeWithoutPrivateWidgets = _.omit(dataTree, privateWidgetNames);
   return treeWithoutPrivateWidgets;
 };
+
+const getDataTreeWithoutSuppressedAutoComplete = (
+  dataTree: DataTree,
+): DataTree => {
+  const metaWidgetIds = Object.keys(dataTree).filter((entityName) => {
```
Comment (ashit-rath) [inline app/client/src/workers/Evaluation/evaluationUtils.ts:719]: `metaWidgetIds` -> `widgetIds` / `entityIds`

**Review by ashit-rath — COMMENTED** [review]
@Tooluloope Remove any code in List widget that deals with private widgets.

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**